### PR TITLE
Define CommonSpaces module

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR extends the `CommonGrids` module with the `CommonSpaces` module, allowing users to construct common spaces. These convenience constructos are a thin layer that simply forward to their convenience grid constructor counterparts and (if necessary) include an additional `staggering` keyword argument.

### CODE REVIEW

**Review by Sbozzolo — COMMENTED** [review]
Code looks great to me.

**Comment by charleskawczynski** [comment]
> Code looks great to me. (Please, add documentation, examples, and entry in the NEWS file)

I did add documentation, but I'd like to figure out a way to minimize the duplicate documentation, since every argument is forwarded except for the `staggering` keyword argument.

For the examples, I added similar counterparts. I kind of wish we could automate this somehow.

File: src/CommonSpaces/CommonSpaces.jl [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    Box3DGrid,
+    SliceXZGrid,
+    RectangleXYGrid
+
+"""
+	ExtrudedCubedSphereSpace(args...; staggering::Staggering, kwargs...)
+
+Calls [`CommonGrids.ExtrudedCubedSphereGrid`](@ref),
+with the args, kwargs, plus an additional `staggering::Staggering`
+keyword, to construct a cell center or cell face space.
```
Comment (Sbozzolo) [inline src/CommonSpaces/CommonSpaces.jl:33]: I suggest we rephrase this along the lines of:

```
Construct an [`ExtrudedFiniteDifferenceSpace`](@ref) for a cubed sphere configuration.

To make the Space face-centered, pass `staggering = Grids.FaceCenter()`.

You can customize various properties of the space, including resolution, number of elements, radius, et cetera, by passing the relative argument. Refer to [`CommonGrids.ExtrudedCubedSphereGrid`](@ref) for the full list of arguments
```

The key point of this function is not that it calls `CommonGrids.ExtrudedCubedSphereGrid`, but that it constructs a `Space`.

**Comment by charleskawczynski** [comment]
Yeah, I'll apply the doc string changes and open an issue to tackle some of the bigger picture docs.

**Comment by charleskawczynski** [comment]
I widened the scope of https://github.com/CliMA/ClimaCore.jl/issues/2068, and updated the API docs, and applied a few doc fixes.

**Review by Sbozzolo — APPROVED** [review]

File: src/CommonSpaces/CommonSpaces.jl [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+ - `quad` the quadrature style (defaults to `Quadratures.GLL{n_quad_points}`)
+ - `h_topology` the horizontal topology (defaults to `Topologies.Topology2D`)
+ - `z_mesh` the vertical mesh, defaults to an `Meshes.IntervalMesh` along `z` with given `stretch`
+ - `enable_bubble` enables the "bubble correction" for more accurate element areas when computing the spectral element space. See [`Grids.SpectralElementGrid2D`](@ref) for more information.
+ - `horizontal_layout_type` the horizontal DataLayout type (defaults to `DataLayouts.IJFH`). This parameter describes how data is arranged in memory. See [`Grids.SpectralElementGrid2D`](@ref) for its use.
+ - `staggering` vertical staggering, can be one of [[`Grids.CellFace`](@ref), [`Grids.CellCenter`](@ref)]
+
+Note that these arguments are all the same
+as [`CommonGrids.Box3DGrid`](@ref),
+except for `staggering`.
```
[the remark this answers is not in the document]
Comment (charleskawczynski) [reply src/CommonSpaces/CommonSpaces.jl:260]: Yeah, I've unfortunately been doing this manually, so I apologize for the inconsistency.

**Pull request merged**